src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -146,100 +198,223 @@
+  public static final String SCHEMA_COMPATIBILITY_CONFIG = "schema.compatibility";
+  private static final String SCHEMA_COMPATIBILITY_DOC =
+      "The schema compatibility rule to use when the connector is observing schema changes. The "
+      + "supported configurations are NONE, BACKWARD, FORWARD and FULL.";
+  private static final String SCHEMA_COMPATIBILITY_DEFAULT = "NONE";
+  private static final String SCHEMA_COMPATIBILITY_DISPLAY = "Schema Compatibility";
 
-  public static final String CONNECT_HDFS_PRINCIPAL_CONFIG = "connect.hdfs.principal";
-  private static final String CONNECT_HDFS_PRINCIPAL_DOC =
-      "The principal to use when HDFS is using Kerberos to for authentication.";
-  public static final String CONNECT_HDFS_PRINCIPAL_DEFAULT = "";
+  public static final String SCHEMA_CACHE_SIZE_CONFIG = "schema.cache.size";
+  private static final String SCHEMA_CACHE_SIZE_DOC =
+      "The size of the schema cache used in the Avro converter.";
+  public static final int SCHEMA_CACHE_SIZE_DEFAULT = 1000;
+  private static final String SCHEMA_CACHE_SIZE_DISPLAY = "Schema Cache Size";
 
-  public static final String CONNECT_HDFS_KEYTAB_CONFIG = "connect.hdfs.keytab";
-  private static final String CONNECT_HDFS_KEYTAB_DOC =
-      "The path to the keytab file for the HDFS connector principal. "
-      + "This keytab file should only be readable by the connector user.";
-  public static final String CONNECT_HDFS_KEYTAB_DEFAULT = "";
+  // Internal group
+  public static final String STORAGE_CLASS_CONFIG = "storage.class";
+  private static final String STORAGE_CLASS_DOC =
+      "The underlying storage layer. The default is HDFS";
+  public static final String STORAGE_CLASS_DEFAULT = "io.confluent.connect.hdfs.storage.HdfsStorage";
+  private static final String STORAGE_CLASS_DISPLAY = "Storage Class";
+
+  private static final String HDFS_GROUP = "hdfs";
+  private static final String HIVE_GROUP = "hive";
+  private static final String SECURITY_GROUP = "security";
+  private static final String CONNECTOR_GROUP = "connector";
+  private static final String INTERNAL_GROUP = "internal";
+  private static final String SCHEMA_GROUP = "schema";
+
+  private static final ConfigDef.Recommender hiveIntegrationDependentsRecommender = new BooleanParentRecommender(HIVE_INTEGRATION_CONFIG);
+  private static final ConfigDef.Recommender hdfsAuthenticationKerberosDependentsRecommender = new BooleanParentRecommender(HDFS_AUTHENTICATION_KERBEROS_CONFIG);
+  private static final ConfigDef.Recommender partitionerClassDependentsRecommender = new PartitionerClassDependentsRecommender();
+  private static final ConfigDef.Recommender schemaCompatibilityRecommender = new SchemaCompatibilityRecommender();
+
+  public static ConfigDef config = new ConfigDef();
+
+  // Define HDFS configuration group
+  static {
+    config.define(HDFS_URL_CONFIG, Type.STRING, "", new HdfsUrlValidator(), Importance.HIGH, HDFS_URL_DOC, HDFS_GROUP, 1, Width.MEDIUM, HDFS_URL_DISPLAY)
```

Comment:
Should this actually be an empty string as a default? Or should it not have a default?